api/v1beta1/types.go

```diff
@@ -544,6 +544,8 @@ const (
 	VMIdentitySystemAssigned VMIdentity = "SystemAssigned"
 	// VMIdentityUserAssigned ...
 	VMIdentityUserAssigned VMIdentity = "UserAssigned"
+	// VMIdentityWorkloadIdentity ...
+	VMIdentityWorkloadIdentity VMIdentity = "WorkloadIdentity"
```

Comment:
when `WorkloadIdentity` type is configured on the VM, do we need to make any changes to the Azure VM configuration?

Reply:
Azure VM configuration -- if you mean the configuration from Azure API point of view, then no. 

We just need a change in azure.json -- and azure.json object is built reading this. 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4138/files#diff-7d084ba6e9e15005850c1b44d72d457cef09f12badee416beda37f957a242364R211

Reply:
>  if you mean the configuration from Azure API point of view, then no.

yes, that's what I meant